### ISSUE TITLE
Porting the ebr-bundle plugin to tycho?

### ISSUE THREAD

**Issue by laeubi** [eclipse/ebr#45]
I wonder if it would be a good idea to port the ebr-bundle-plugin to tycho (there are already some additional tools mojos in tycho-extras).

I would see the following benefits:

1. user do not need to build the plugin itself before starting to use ebr
2. because its integrated in the usual release cycle no incompatibilities with newer tycho version (or at least they will be discovered very early)
3. tighter integration possible, currently its a two-step-process we should check how this can be integrated more smooth to make this work in one go.

**Comment by guw** [comment]
The goal of the EBR project is to provide recipes for making non-OSGi jars OSGi compatible. Most of the time that only requires metadata. Occasionally it may require Java code (eg., an activator, providing/integrating with services/class loading). The EBR Maven plug-ins support that, i.e. it's possible to patch/override/replace code of legacy jars. Additionally, it's possible to combine multiple jars into one ("all-in-one") or producing only a subset of a jar (eg., to remove IP issues).

However, nothing at EBR mandates to use a specific technology. If Tycho provides the same functionality we can start using that and don't need our own plug-ins. 

Ultimately, I would love to see everything running in a single refactor and being used interchangeably. The deep integration with Tycho is possible. We have a a plug-in stub for extending Tycho.

We are not too far, though. I believe the remaining issue is to refactor Tycho's target platform resolution into a later phase, which allows arbitrary projects to produce OSGi artifacts for Tycho to resolve.

**Comment by laeubi** [comment]
> However, nothing at EBR mandates to use a specific technology. If Tycho provides the same functionality we can start using that and don't need our own plug-ins.

Thats why I think it might be good to port that "maven-plugin-part" (not EBR as a whole(!)) to tycho as at least this part where specific to tycho-technology.

I'm currently investigation (if there is some free time) the part of target platform resolution and the ebr plugin seems a good example and thus the idea to maybe first port it to be part of tycho itself :-)

@mickaelistria wdyt?

**Comment by mickaelistria** [comment]
@laeubi I think that's a good idea, however, can you please bring this proposal to the tycho-dev mailing-list so all committers can voice their opinion and ideas?

**Comment by jonahgraham** [comment]
Closing out old issues that are not likely to be addressed going forward. https://github.com/eclipse/ebr#history
